compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -1086,7 +1074,6 @@ class Typer extends Namer with TypeAssigner with Applications with Implicits wit
   }
 
   def typedQuote(tree: untpd.Quote, pt: Type)(implicit ctx: Context): Tree = track("typedQuote") {
-    ctx.compilationUnit.containsQuotes = true
```

Comment:
We do need `ctx.compilationUnit.containsQuotesOrSplices = true` in case there are only quotes in the tree. I can add the fix with the test case in #3662.